### PR TITLE
feeding-sleeping-playing-implementation

### PR DESCRIPTION
**Pull request by 1ukasz**

### CODE REVIEW

File: js/game.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ export default class Game {
     this.hungerIntervalId = setInterval(this.decreaseHunger, 1000);
     this.energyIntervalId = setInterval(this.decreaseEnergy, 2000);
     this.funIntervalId = setInterval(this.decreaseFun, 1000);
+    this.isEating = false;
```
Comment (Sumick) [inline js/game.js:10]: **question:** Why did you need to add these 3 flags here? Couldn't you solve it using state only?

File: js/game.js [hunk 2]
```diff
@@ -99,48 +102,74 @@ export default class Game {
...
-      this.btnSingleListener(
-        `-${this.tamagotchi.states.eating}`,
-        this.tamagotchi.states.eating.toUpperCase()
-      );
-      this.tamagotchi.hunger.value += 2;
-      this.displayTimeout(
-        this.tamagotchi.displayHunger,
-        this.tamagotchi.hunger.element
-      );
+      if (!this.isEating) {
```
Comment (Sumick) [inline js/game.js:107]: **question:** Why not to ```this.tamagotchi.state !== this.tamagotchi.states.eating```?

File: js/game.js [hunk 3]
```diff
@@ -99,48 +102,74 @@ export default class Game {
...
+        this.isEating = true;
```
Comment (Sumick) [inline js/game.js:108]: **suggestion:** Then you could remove these flags, and reduce code complexity

File: js/game.js [hunk 4]
```diff
@@ -99,48 +102,74 @@ export default class Game {
...
+        this.isSleeping = false;
+        this.isPlaying = false;
+        clearInterval(this.sleepingIntervalId);
+        clearInterval(this.playingIntervalId);
+        this.btnSingleListener(
+          `-${this.tamagotchi.states.eating}`,
+          this.tamagotchi.states.eating.toUpperCase()
```
Comment (Sumick) [inline js/game.js:115]: **suggestion:** You can call `.toUpperCase()` inside `btnSingleListener` function, so you would not have to call it every time you pass the state

File: js/game.js [hunk 5]
```diff
@@ -99,48 +102,74 @@ export default class Game {
...
-      this.btnSingleListener(
-        `-${this.tamagotchi.states.sleeping}`,
-        this.tamagotchi.states.sleeping.toUpperCase()
-      );
-      this.tamagotchi.energy.value += 2;
-      this.displayTimeout(
-        this.tamagotchi.displayEnergy,
-        this.tamagotchi.energy.element
-      );
+      if (!this.isSleeping) {
```
Comment (Sumick) [inline js/game.js:129]: **question:** Why not to ```this.tamagotchi.state !== this.tamagotchi.states.sleeping```?

**Review by Sumick — APPROVED** [review]
Good job! Im approving, but leaving you few non blocking comments, that you could use for optimizing your implementation.

File: js/game.js (hunk 2 above)
Comment (1ukasz) [reply js/game.js:107]: Good point. I'll fix it in next PR.

File: js/game.js (hunk 3 above)
Comment (1ukasz) [reply js/game.js:108]: Good point. I'll fix it in next PR.

File: js/game.js (hunk 5 above)
Comment (1ukasz) [reply js/game.js:129]: Good point. I'll fix it in next PR.

File: js/game.js (hunk 1 above)
Comment (1ukasz) [reply js/game.js:10]: Good point. I'll fix it in next PR.
